src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -101,16 +127,75 @@ export default class VizShonenJump extends Connector {
             });
     }
 
+    async _getMangaVolumes(manga) {
+        let auth = await this._getUserInfo();
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request);
+
+        return [...data.querySelectorAll('#o_products tr td:last-of-type a')]
+            .filter(element => {
+                // login required (e.g. age verification) => javascript:void('login to read');
+                if(/(?=javascript:)(?=.*login)/i.test(element.href)) {
+                    return auth.isLoggedIn && auth.isAdult;
+                }
+                // subscription required => javascript:void('join to read');
+                if(/(?=javascript:)(?=.*join)/i.test(element.href)) {
+                    return auth.isMember;
+                }
+                // free
+                return true;
+            })
+            .map(chapter => {
+                return {
+                    id: chapter.href.replace("hakuneko://cache/", ""),
+                    title: 'Vol. ' + chapter.href.match(/-volume-([-_0-9]+)/)[1]
+                };
+            });
+    }
+
+    async _getChapters(manga) {
+        if (manga.id.startsWith("/shonenjump/chapters")) {
+            return await this._getMangaChapters(manga);
+        } else if (manga.id.startsWith("/library")) {
+            return await this._getMangaVolumes(manga);
+        }
```

Comment:
Missing return (method will await forever), consider to throw if non of the conditions above is fulfilled, e.g.
```javascript
throw new Error(`Failed to get chapters/volumes for manga ${manga.id}, because the manga type is not supported!`);
```